src/components/vm/consoles/consoles.jsx

```diff
@@ -214,34 +257,78 @@ export const ConsoleCard = ({ state, vm, config, onAddErrorNotification, isExpan
         );
     }
 
-    actions.push(
-        <Button
-            key="expand-compress"
-            variant="link"
-            onClick={() => {
-                const urlOptions = { name: vm.name, connection: vm.connectionName };
-                const path = isExpanded ? ["vm"] : ["vm", "console"];
-                return cockpit.location.go(path, { ...cockpit.location.options, ...urlOptions });
-            }}
-            icon={isExpanded ? <CompressIcon /> : <ExpandIcon />}
-            iconPosition="right">{isExpanded ? _("Compress") : _("Expand")}
-        </Button>
+    if (!isStandalone) {
+        actions.push(
+            <Button
+                key="expand-compress"
+                variant="link"
+                onClick={() => {
+                    const urlOptions = { name: vm.name, connection: vm.connectionName };
+                    const path = isExpanded ? ["vm"] : ["vm", "console"];
+                    return cockpit.location.go(path, { ...cockpit.location.options, ...urlOptions });
+                }}
+                icon={isExpanded ? <CompressIcon /> : <ExpandIcon />}
+                iconPosition="right">{isExpanded ? _("Compress") : _("Expand")}
+            </Button>
+        );
+    }
+
+    if (!isStandalone && type == "graphical") {
+        const urlOptions = { name: vm.name, connection: vm.connectionName };
+        const path = ["vm", "vnc"];
+        const href = "#" + cockpit.location.encode(path, { ...cockpit.location.options, ...urlOptions });
+
+        actions.push(
+            <Button
+                key="detach"
+                variant="link"
+                component="a"
+                href={href}
+                target="_blank"
+                onClick={(event) => {
+                    let header_height = 53;
+                    const title = document.querySelector(`#${vmId(vm.name)}-consoles .pf-v6-c-card__header-main`);
+                    if (title) {
+                        // The 8 below is the padding of the expanded version and fixed in consoles.css
+                        header_height = 8 + title.offsetHeight + 8;
+                    }
+                    const sz = lastVncRemoteSize.current;
+                    window.open(href, "vnc-" + vm.name, `popup,width=${sz[0]},height=${sz[1] + header_height}`);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2254-9e3d340a-20250805-063134-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)